webapp/channels/src/components/with_tooltip/create_tooltip.tsx

```diff
@@ -3,55 +3,93 @@
 
 import React from 'react';
 import type {ComponentProps} from 'react';
-import type {MessageDescriptor} from 'react-intl';
+import {type MessageDescriptor} from 'react-intl';
 
 import RenderEmoji from 'components/emoji/render_emoji';
-import {ShortcutKey, ShortcutKeyVariant} from 'components/shortcut_key';
 import Tooltip from 'components/tooltip';
 
-import {getStringOrDescriptorComponent} from './utils';
+import {TooltipShortcutSequence, type ShortcutDefinition} from './shortcut';
+import {getAsFormattedMessage} from './utils';
+
+type EmojiStyle = 'inline' | 'large' | undefined;
 
 export type CommonTooltipProps = {
     id: string;
-    title: string | MessageDescriptor;
-    hint?: string | MessageDescriptor;
-    shortcut?: string[];
+    title: string | MessageDescriptor | React.ReactElement;
+    hint?: string | MessageDescriptor | React.ReactElement;
+    shortcut?: ShortcutDefinition;
     emoji?: string;
+    emojiStyle?: EmojiStyle;
 }
 
 export function createTooltip(commonTooltipProps: CommonTooltipProps) {
     return (props: Omit<ComponentProps<typeof Tooltip>, 'children' | 'id'>) => {
-        const title = getStringOrDescriptorComponent(commonTooltipProps.title);
-        const hint = getStringOrDescriptorComponent(commonTooltipProps.hint);
-
-        const emoji = commonTooltipProps.emoji && (
-            <RenderEmoji
-                emojiName={commonTooltipProps.emoji}
-                size={12}
-            />
-        );
+        const contents = [];
+
+        if (commonTooltipProps.emoji && commonTooltipProps.emojiStyle === 'large') {
+            // HARRISONTODO The vertical padding in the designs is larger than this, but it isn't final yet
```

Comment:
Is this TODO already handled?

Reply:
I forgot to add Abhijit for a UX review since I meant to ask him that. I'll make sure to remove this before the PR is merged